plugins/GCodeWriter/GCodeWriter.py

```diff
@@ -68,23 +68,21 @@ def _serialiseSettings(self, settings):
         prefix = ";SETTING_" + str(GCodeWriter.version) + " "  # The prefix to put before each line.
         prefix_length = len(prefix)
 
-        all_settings = InstanceContainer("G-code-imported-profile") #Create a new 'profile' with ALL settings so that the slice can be precisely reproduced.
-        all_settings.setDefinition(settings.getBottom())
-        for key in settings.getAllKeys():
-            all_settings.setProperty(key, "value", settings.getProperty(key, "value")) #Just copy everything over to the setting instance.
-        serialised = all_settings.serialize()
+        global_stack = Application.getInstance().getGlobalContainerStack()
+        container_with_profile = global_stack.findContainer({"type": "quality"})
+        serialized = container_with_profile.serialize()
```

Comment:
This only includes the "quality" profile settings and omits the "user" profile settings. So any settings the user has made that are not stored in the quality profile will not be stored with the gcode profile. We should have the settings from both "quality" and "user" combined into one.